### PR TITLE
Create coding_standard.md

### PR DESCRIPTION
**Pull request by hnwyllmm**
### Task Description

There is no coding standard file in the project.

### Solution Description

I copy the coding standard file from oceanbase org.

### CODE REVIEW

File: docs/coding_standard.md [hunk 1]
```diff
@@ -0,0 +1,2569 @@
...
+- Set the session log level variable
+- Set tenant log level variable
+- Set syslog level variable
+In addition, version 1.0 also supports the concepts of log modules and submodules. When printing logs in the program, it is necessary to indicate the module (or module + submodule) to which the log belongs and the log printing level to which the log belongs. The system supports users to set different printing levels for each module and sub-module.
+**Stricter log printing format**
+In version 0.5, there is a problem that the print log format is not uniform and difficult to read. For example, when printing the variable `m` whose value is 5, there are many different printing formats: "m = 5", "m=5", "m(5)",
+"m is 5", "m:5", etc. The new log module allows users to print the values of required variables in the form of key-value pairs.
+## 11.1 Log Printing Level
+
+| Log Level | Level Definition |
```
Comment (caifeizhi) [inline docs/coding_standard.md:2435]: we have more log level like `WDIAG`, you may need to  check and update it

**Comment by caifeizhi** [comment]
Invite @yangzhifeng to review

File: docs/coding_standard.md [hunk 2]
```diff
@@ -0,0 +1,2569 @@
...
+  static const int CONST_V;
+}
+// in the implenmentation file
+const int ObBar::CONST_V = 1;
+```
+
+Before C\+\+11, the C\+\+98 standard only allowed static const variables of intergral type to be initialized with definitions included in the class declaration. In C++11, constexpr is introduced, and static constexpr member variables (including types such as double) can also be initialized in the declaration. This kind of variable will not generate static area storage after compilation.
+
+```txt
+Before C++11, the values of variables could be used in constant expressions only
```
[the remark this answers is not in the document]
Comment (hnwyllmm) [reply docs/coding_standard.md]: I use '>' instead of '```' to represent the txt.

File: docs/coding_standard.md [hunk 3]
```diff
@@ -0,0 +1,2571 @@
+
+| Number | Document Version | Revised Chapter | Reason for Revision | Revision Date   |
+| -------| ---------------- | --------------- | ------------------- | -------------   |
+| 1      | 1.0              |                 | New                 | June 15th, 2023 |
+
+# 1 Introduction
+This coding standard is applicable to the OceanBase project of Ant Group. It provides some coding constraints and defines coding styles. In the OceanBase project, test code must adhere to the coding style defined in this document, and it is recommended that test code also adhere to the coding constraints defined in this document. Other code must adhere to both the coding constraints and coding style defined in this document.
```
Comment (yangzhifeng) [inline docs/coding_standard.md:7]: test code ？

**Review by caifeizhi — APPROVED** [review]
LGTM

**Pull request closed without merging**